### PR TITLE
feat(slack): typeahead support for assignment dropdown

### LINKED ISSUE #62899: Add typeahead support for assignment dropdown

### PR DESCRIPTION
**Pull request by isabellaenriquez**
This PR adds the `SlackOptionsLoadEndpoint`, allowing for typeahead support for Slack issue alert assignment dropdowns.

https://github.com/getsentry/sentry/assets/45607721/bbacf639-80e2-4c1b-836b-b15cd8950108

### CODE REVIEW

File: src/sentry/integrations/slack/message_builder/issues.py [hunk 1]
```diff
@@ -434,7 +434,7 @@ def build(self, notification_uuid: str | None = None) -> Union[SlackBlock, Slack
             ):
                 actions.append(self.get_button_action(action))
             elif action.name == "assign":
-                actions.append(self.get_static_action(action))
+                actions.append(self.get_external_select_action(action))
```
Comment (isabellaenriquez) [inline src/sentry/integrations/slack/message_builder/issues.py:437]: this point is only reached if the block kit flag is enabled

**Comment by ceorourke** [comment]
This looks great! Can you add some tests?

File: src/sentry/integrations/slack/requests/options_load.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        if "payload" not in self.request.data:
+            raise SlackRequestError(status=status.HTTP_400_BAD_REQUEST)
+
+        try:
+            self._data = json.loads(self.data["payload"])
+        except (KeyError, IndexError, TypeError, ValueError):
+            raise SlackRequestError(status=status.HTTP_400_BAD_REQUEST)
+
+        if self.data.get("type", None) not in VALID_PAYLOAD_TYPES:
```
Comment (ceorourke) [inline src/sentry/integrations/slack/requests/options_load.py:38]: nit: `None` is the default value for `.get()` so you don't need to specify it

**Review by ceorourke — APPROVED** [review]
lgtm, just need to fix up that test

**Review by ceorourke — APPROVED** [review]

**Pull request merged**